packages/Uploader/src/Uploader.js

```diff
@@ -0,0 +1,219 @@
+const defaultProps = {
+  a11yText: null,
+  acceptableFileTypes: ["*/*"],
+  allowMultipleFile: true,
+  defaultIsDisable: false,
+  hasAutoupload: true,
+  headers: [],
+  isBodyDroppableArea: true,
+  maximumFileSize: oneMebibyte * 10, // 1048576bytes * 10 = 10,485,760 Mebibytes
+  onChange: () => {},
+  onFinished: () => {},
+};
+
+function getDocumentBody() {
+  return document.body;
+}
+
+function getContainer(refContainer) {
+  return function findContainer() {
+    return refContainer.current;
+  };
+}
+
+function UploaderComponent(props, ref) {
+  const {
+    a11yText,
+    acceptableFileTypes,
+    allowMultipleFile,
+    hasAutoupload,
+    maximumFileSize,
+    onChange,
+    onFinished,
+    isBodyDroppableArea,
+    endpoint,
+    children,
+    defaultIsDisable,
+    headers,
+  } = props;
+
+  const refInput = React.useRef();
+  const refContainer = React.useRef(null);
+  const [refId] = React.useState(uuidv4());
+  const i18n = useI18n();
+  const label = a11yText || i18n.t("uploader.label");
+
+  React.useImperativeHandle(ref, () => ({
+    focus: () => {
+      refInput.current.focus();
+    },
+  }));
+
+  const { files, hasFinished, isDisabled, removeFile, cancelFile, setFiles, upload } = useProcessFiles({
+    defaultIsDisable,
+    endpoint,
+    hasAutoupload,
+    headers,
+    onChange,
+    onFinished,
+  });
+
+  function handleChange(event) {
+    if (isDisabled) return;
+
+    const files = getFiles({ event, maximumFileSize, acceptableFileTypes, endpoint });
+    setFiles(() => {
+      if (refInput.current) {
+        refInput.current.value = "";
+      }
+      return allowMultipleFile ? files : [files[0]]; // in case only allow one file per upload
+    });
+  }
+
+  const { isDragLeave, isDragOver } = useDragAndDropEvents({
+    dropArea: isBodyDroppableArea ? getDocumentBody : getContainer(refContainer),
+    handleChange,
+    defaultIsDisable,
+  });
+
+  function FileInput(props) {
+    const { children } = props;
+    return (
+      <div css={containerStyles} ref={refContainer}>
+        <input
+          multiple={allowMultipleFile}
+          id={refId}
+          onChange={handleChange}
+          ref={refInput}
+          css={inputFileStyles}
+          type="file"
+          accept={acceptableFileTypes.join(",")}
+        />
+        <label css={labelStyles} htmlFor={refId}>
+          <span css={stylers.visuallyHidden}>{label}</span>
+        </label>
+        {/* a11y help require, is this the best approach? */}
+        {/* aria-hidden will prevent from rendering content that can be counter intuitive for the screen reader */}
+        <div aria-hidden>{children}</div>
+      </div>
+    );
+  }
+
+  return (
+    <>
+      {children({
+        FileInput,
+        files,
+        hasFinished,
+        isDisabled,
+        isDragLeave,
+        isDragOver,
+        removeFile,
+        cancelFile,
+        upload,
+      })}
+    </>
+  );
+}
+
+const Uploader = React.forwardRef(UploaderComponent);
```

Comment:
Seems cleaner to do this inline above, no need for a Uploader AND UploaderComponent that both need propTypes attached to them.